VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimLettersScreen/ClaimLettersScreen.tsx

```diff
@@ -36,29 +39,48 @@ const ClaimLettersScreen = ({ navigation }: ClaimLettersScreenProps) => {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
   const dispatch = useAppDispatch()
-  const { loading, decisionLetters, downloading } = useSelector<RootState, DecisionLettersState>(
-    (state) => state.decisionLetters,
-  )
+  const queryClient = useQueryClient()
   const claimsInDowntime = useDowntime(DowntimeFeatureTypeConstants.claims)
   const prevScreen = useNavigationState((state) => state.routes[state.routes.length - 2]?.name)
-
+  const { demoMode } = useSelector<RootState, DemoState>((state) => state.demo)
+  const [letterID, setLetterID] = useState<string>('')
+  const { data: decisionLettersData, isLoading: loading } = useDecisionLetters({
+    enabled: screenContentAllowed('WG_ClaimLettersScreen') && !claimsInDowntime,
+  })
+  const {
+    isLoading: downloading,
+    isError: downloadLetterError,
+    error: downloadLetterErrorDetails,
+    refetch: refetchLetter,
+  } = useDownloadDecisionLetter(letterID, demoMode, {
+    enabled: letterID.length > 0,
+  })
   // This screen is reachable from two different screens, so adjust back button label
+  const decisionLetters = decisionLettersData?.data || ([] as DecisionLettersList)
   const backLabel = prevScreen === 'ClaimDetailsScreen' ? t('claimDetails.title') : t('claims.title')
 
-  const snackbarMessages: SnackbarMessages = {
-    successMsg: '',
-    errorMsg: t('claimLetters.download.error'),
+  const fetchInfoAgain = () => {
+    queryClient.invalidateQueries({ queryKey: decisionLettersKeys.decisionLetters })
```

Comment:
Instead of invalidating the query, we should probably call the `refetchLetter` function instead. Invalidating the query doesn't guarantee the info will be fetched right away, especially because our cache stale time is 5 seconds. This function wouldn't even be needed, and you could just pass `refetchLetter` to the error component

Reply:
updated